galaxy_ng/app/tasks/publishing.py

```diff
@@ -25,54 +42,67 @@ def import_and_move_to_staging(artifact_pk, **kwargs):
     This task will not wait for the enqueued tasks to finish.
     """
     inbound_repository_pk = kwargs.get('repository_pk')
-    import_collection(artifact_pk=artifact_pk, repository_pk=inbound_repository_pk)
-
-    content_artifact = ContentArtifact.objects.get(artifact_id=artifact_pk)
-    collection_version = content_artifact.content.ansible_collectionversion
+    import_collection(temp_file_pk=temp_file_pk,
+                      repository_pk=inbound_repository_pk)
 
-    # enqueue task to add collection_version to staging repo
     try:
-        staging_repo = AnsibleRepository.objects.get(name=STAGING_NAME)
+        staging_repo = AnsibleDistribution.objects.get(name=STAGING_NAME).repository
     except AnsibleRepository.DoesNotExist:
         raise RuntimeError(f'Could not find staging repository: "{STAGING_NAME}"')
-    locks = [staging_repo]
-    task_args = (collection_version.pk, staging_repo.pk)
-    enqueue_with_reservation(add_content_to_repository, locks, args=task_args)
 
-    # enqueue task to remove collection_verion from inbound repo
     inbound_repo = AnsibleRepository.objects.get(pk=inbound_repository_pk)
-    locks = [inbound_repo]
-    task_args = (collection_version.pk, inbound_repository_pk)
-    enqueue_with_reservation(remove_content_from_repository, locks, args=task_args)
 
+    remove_locks = [staging_repo]
+    add_locks = [inbound_repo]
```

Comment:
I think these two values need to be switched

Reply:
It certainly looks wrong now. Testing it again now to see if I switch it for a reason.